addons/website/static/src/js/http_cookie.js

```diff
@@ -0,0 +1,18 @@
+/** @odoo-module **/
+
+import cookieUtils from 'web.utils.cookies';
+
+const originFunc = cookieUtils.checkCookie;
+cookieUtils.checkCookie = (type, name) => {
+    const result = originFunc.apply(cookieUtils, [type, name]);
+    if (result && type === 'optional') {
+        const consents = JSON.parse(cookieUtils.get_cookie('accepted_cookie_types') || '{}');
+        if ('optional' in consents) {
+            return consents['optional'];
+        }
+        return !document.getElementById('cookies-consent-essential');
```

Comment:
If the goal of this line is actually checking if the cookies bar setting is disabled on the website, should be commented, not straightforward